test/echoes_to_frame_converter_task_test.rb

```diff
@@ -223,11 +234,7 @@ def create_configure_task
     end
 
     def write_pose(task, pose)
-        expect_execution do
-            syskit_write task.sensor2ref_pose_port, pose
-        end.to do
-            have_one_new_sample(task.frame_port)
-        end
+        expect_execution { syskit_write task.sensor2ref_pose_port, pose }.to_run
```

Comment:
This is a heisenbug, actually. There is no guarantee in all the write_pose/write_echo pairs that the tests have that the new pose would have arrived before the new echo.

Reply:
In addition, I don't agree with your analysis that only echo is meaningful on the port driven. The image output does change on pose change.